### DISCUSSION TITLE
Question: How to implement this: UserRequest inside InlineKeyboard

### DISCUSSION THREAD

**Discussion by voydninja** [python-telegram-bot/python-telegram-bot#4442] [Q&A] [answered]
I just found this https://core.telegram.org/bots/features#chat-and-user-selection that says it's possible to use UserRequest inside InlineMarkup, but It seems impossible.
Is this right? or just for ad?

**Comment by harshil21** [comment]
why do you think it's impossible?

**Reply by voydninja** [reply]
I tried 10+ times, manual googling and gpt assistant. Couldn't find ANY solution.
Is this possible with python-telegram-bot?
Basically, I wanna implement this flow in my bot:
#1 User clicks inlineButton and it shows user selection panel.
#2 User chooses other users he's chatting with on that selection box.
#3 Bot sends hello message to users he chose.
I am not sure this is possible...🧛‍♂️

**Reply by harshil21** [reply]
Just use https://docs.python-telegram-bot.org/en/doc-fixes/telegram.keyboardbutton.html#telegram.KeyboardButton.params.request_users. you can only do point 3. if the user has contacted the bot before.

**Reply by voydninja** [reply]
Thanks @harshil21 
But I want to use request_users with InlineKeyboard not Keyboardbutton.
Is this possible?
As we can see through https://core.telegram.org/bots/features#chat-and-user-selection, it uses inlinebutton not a keyboardbutton.

**Reply by harshil21** [reply, marked as answer]
that's a mistake on their graphic. Their documentation clearly writes it to be a KeyboardButton

**Reply by voydninja** [reply]
Oh, yeah. You are right. Thanks for you help. @harshil21
